### PR TITLE
Enable sia for ase

### PR DESCRIPTION
**Pull request by hforeste**
- Show correlated detectors to downtimes detected by runtime availability detector
- Adjust routes to reflect aseAvailabilityAnalysis and to run Sia
- Abstract the different type of resource results from the Sia module 
- Abstract the AseService and the SiteService from SiaCtrl, DowntimeTimelineCtrl

image

### CODE REVIEW

File: AppLensV2/app/AppServiceEnvironment/AppServiceEnvironmentCtrl.ts [hunk 1]
```diff
@@ -49,6 +49,27 @@ module SupportCenter {
                 self.DetectorsService.getDetectors(self.hostingEnvironment).then(function (data: DetectorDefinition[]) {
                     self.detectors = data;
                     self.detectorListLoaded = true;
+                    self.SiaService.getSiaResponse().then(function (data: IAnalysisResult) {
+                        var siaResponse = data.Response;
+                        _.each(siaResponse.NonCorrelatedDetectors, function (item: DetectorDefinition) {
+                            _.each(self.DetectorsService.detectorsList, function (detector: DetectorDefinition) {
+                                if (item.DisplayName == detector.DisplayName) {
```
Comment (ShekharGupta1988) [inline AppLensV2/app/AppServiceEnvironment/AppServiceEnvironmentCtrl.ts]: use ===

File: AppLensV2/app/AppServiceEnvironment/AppServiceEnvironmentCtrl.ts [hunk 2]
```diff
@@ -49,6 +49,27 @@ module SupportCenter {
...
+                                    detector.Correlated = 0;
+                                }
+                            });
+                        });
+
+                        _.each(siaResponse.Payload, function (item: AnalysisData) {
+                            _.each(self.DetectorsService.detectorsList, function (detector: DetectorDefinition) {
+                                if (item.DetectorDefinition.DisplayName == detector.DisplayName) {
```
Comment (ShekharGupta1988) [inline AppLensV2/app/AppServiceEnvironment/AppServiceEnvironmentCtrl.ts]: same here

Comment (ShekharGupta1988) [inline AppLensV2/app/AppServiceEnvironment/AppServiceEnvironmentCtrl.ts]: ===

File: AppLensV2/app/app.ts [hunk 3]
```diff
@@ -136,6 +138,19 @@ module SupportCenter {
                         analysisType: 'perfAnalysis'
                     }
                 })
+                .state('home3.aseAvailabilityAnalysis', {
```
Comment (stevenernst131) [inline AppLensV2/app/app.ts]: I am not a fan of "home3", would prefer something more descriptive but I am also fine if you want to keep it.

**Comment by hforeste** [comment]
thanks guys

**Pull request merged**